// Copyright 2024 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package secretflow.serving;

message TraceLogConfig {
  // defaule value: "trace.log"
  string trace_log_path = 1;
  // Byte. default value: 500 * 1024 * 1024 (500MB)
  int32 max_trace_log_file_size = 2;
  // default value: 10
  int32 max_trace_log_file_count = 3;
}

// Serving trace config options
message TraceConfig {
  // dump trace to a logfile, default: false
  bool trace_log_enable = 1;
  // if trace_log_enable is true, use this log config to dump trace info
  TraceLogConfig trace_log_conf = 2;
}
